### PR TITLE
GitHub action and consistency changes

### PR DESCRIPTION
**Pull request by tcekle**
## Changes

* Add release build action
* Separate CI and PR builds
* Fix docs to match implementation of `MachineType` ( [AB#2301](https://dev.azure.com/dataio/3ebf86f1-ed57-4e86-83a7-f7a9fa811b76/_workitems/edit/2301) )
* Add script to push to Nuget.org
* Rename `PlacePart` to `PlacePartMessage` for consistency

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,9 +4,6 @@ on:
   push:
     branches:
     - main
-  pull_request:
```
Comment (tcekle) [inline .github/workflows/ci.yml:7]: Separate CI and PR so we don't push to nuget on every PR

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -15,18 +12,24 @@ jobs:
         working-directory: ./src
     steps:
     - name: Checkout
-      uses: actions/checkout@v2
+      uses: actions/checkout@v4
```
Comment (tcekle) [inline .github/workflows/ci.yml:15]: We were getting warnings about these actions, so decided to go through them all and update

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: Pull Request
```
Comment (tcekle) [inline .github/workflows/pr.yml:1]: Same as CI, just no pushing to nuget

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - '*.*.*'
```
Comment (tcekle) [inline .github/workflows/release.yml:6]: Will run if we push a tag (such as `v3.0.7`)

File: docs/help/api/mqtt/mqtt-automated-handler.md [hunk 5]
```diff
@@ -248,10 +248,10 @@ Level           | Description
 `xhsessionid`   | The session ID for the current instance of xH700 software.
 
 #### Fields
-Name                   | Description
------------------------|------------------------------------------------------------------------------------
-`Active` (`bool`)      | Always true.
-`MachineType` (`enum`) | The [`MachineType`](#machinetype) representing the type of machine that is running.
+Name                     | Description
```
Comment (tcekle) [inline docs/help/api/mqtt/mqtt-automated-handler.md:251]: Match the ConneX documentation

File: src/DataIO.ConneX.Api.Mqtt/Messages/PsvSystem/PlacePartMessage.cs [hunk 6]
```diff
@@ -7,7 +7,7 @@ namespace DataIO.ConneX.Api.Mqtt.Messages.PsvSystem
     /// <summary>
     /// Class representing a Place Part MQTT message
     /// </summary>
-    public class PlacePart
+    public class PlacePartMessage
```
Comment (tcekle) [inline src/DataIO.ConneX.Api.Mqtt/Messages/PsvSystem/PlacePartMessage.cs:10]: Changed for consistency with other messages

**Review by tcekle — COMMENTED** [review]
@villanuevan @bplasota 

Some changes to the build to finish AB#1445

File: src/push.ps1
Comment (bplasota) [inline src/push.ps1:1]: I feel like I'm back in the early 90s with the old school C and C++ guy at my work that didn't want to let go of the '{" bracket on the same line. It was three dev including me telling him to put it on a separate line vs his way. He never changed.

**Review by bplasota — APPROVED** [review]
It all looks fine to me.

**Comment by villanuevan** [comment]
LGTM

**Pull request merged**